### PR TITLE
Added toggleable auto travel mode

### PR DESCRIPTION
**Pull request by ZhilkinSerg**
#### Summary

SUMMARY: Features "Added toggleable auto travel mode"

#### Purpose of change

Allow faster traversing of maps with automatic obstacle avoiding.

#### Describe the solution

Added new action (no default keybinding set) which would toggle auto travel mode - player will try to move in selected direction avoiding visible obstacles.

#### Additional context

City:

auto_travel_mode

Forest:

auto_travel_mod_forest

### CODE REVIEW

File: src/handle_action.cpp [hunk 1]
```diff
@@ -2082,6 +2041,36 @@ bool game::handle_action()
...
                 break;
         }
+        if( dx != 0 || dy != 0 ) {
+            if( !u.get_value( "remote_controlling" ).empty() &&
+                ( u.has_active_item( "radiocontrol" ) || u.has_active_bionic( bio_remote ) ) ) {
+                rcdrive( dx, dy );
+            } else if( veh_ctrl ) {
+                pldrive( dx, dy );
+            } else {
+                if( auto_travel_mode/* && safe_mode == SAFE_MODE_ON*/ ) {
```
Comment (Night-Pryanik) [inline src/handle_action.cpp]:
```suggestion
                if( auto_travel_mode ) {
```

Comment (ZhilkinSerg) [reply src/handle_action.cpp]: Thanks.

After a few tests I am considering making auto travel mode possible only when safe mode is enabled. I will test some more and either remove commented code or uncomment it.

**Comment by esotericist** [comment]
Holy crap this is really cool and I had no idea this was being worked on.

**Comment by DaviBones** [comment]
Was this implemented correctly for circular distances too, where deciding whether to sidestep high movement cost tiles is less trivial?

**Comment by ZhilkinSerg** [comment]
> 
> 
> Was this implemented correctly for circular distances too, where deciding whether to sidestep high movement cost tiles is less trivial?

New feature uses original pathfinding routines. Read their code and report if you find anything suspicious.

**Comment by kevingranade** [comment]
This pull request has been mentioned on **Cataclysm: Dark Days Ahead**. There might be relevant details there:

https://discourse.cataclysmdda.org/t/can-you-auto-walk-in-a-line-like-in-dungeon-crawl/27462/4

**Pull request merged**